src/ERC6900/plugins/MultipleOwnerPlugin.sol

```diff
@@ -0,0 +1,350 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {UserOperation} from "@ERC4337/interfaces/UserOperation.sol";
+
+import {EtherspotWalletV2, UUPSUpgradeable} from "../wallet/EtherspotWalletV2.sol";
+import {ArrayLib} from "../libraries/ArrayLib.sol";
+import {ErrorsLib} from "../libraries/ErrorsLib.sol";
+import {IMultipleOwnerPlugin} from "../interfaces/IMultipleOwnerPlugin.sol";
+
+import {ManifestFunction, ManifestAssociatedFunctionType, ManifestAssociatedFunction, PluginManifest, ManifestExecutionFunction} from "@ERC6900/src/interfaces/IPlugin.sol";
+import {IStandardExecutor} from "@ERC6900/src/interfaces/IStandardExecutor.sol";
+import {BasePlugin} from "@ERC6900/src/plugins/BasePlugin.sol";
+
+contract MultipleOwnerPlugin is BasePlugin, IMultipleOwnerPlugin {
+    using ECDSA for bytes32;
+
+    string public constant NAME = "Multiple Owner Plugin";
+    string public constant VERSION = "1.0.0";
+    string public constant AUTHOR = "Etherspot";
+
+    uint256 internal constant _SIG_VALIDATION_PASSED = 0;
+    uint256 internal constant _SIG_VALIDATION_FAILED = 1;
+
+    // bytes4(keccak256("isValidSignature(bytes32,bytes)"))
+    bytes4 internal constant _1271_MAGIC_VALUE = 0x1626ba7e;
+
+    mapping(address => address[]) internal _multipleOwners;
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Execution functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc IMultipleOwnerPlugin
+    function addOwner(address _newOwner) external {
+        if (isOwner(_newOwner)) revert ErrorsLib.AlreadyAnOwner();
+        _addOwner(_newOwner);
+        emit OwnerAdded(msg.sender, _newOwner);
+    }
+
+    /// @inheritdoc IMultipleOwnerPlugin
+    function removeOwner(address _owner) external {
+        if (!isOwner(_owner)) revert ErrorsLib.NotAnOwner();
+        _removeOwner(_owner);
+        emit OwnerRemoved(msg.sender, _owner);
+    }
+
+    /// @inheritdoc IMultipleOwnerPlugin
+    function transferOwnership(address newOwner) external {
+        _transferOwnership(newOwner);
+    }
+
+    /// This function is an issue with using IERC1271.
+    /// It can work using ECDSA.recover to obtain the signer from the digest
+    /// and pass that into isValidSignatureNow call.
+    /// However, when signing from an SCW, this fails due to ECDSA for SCWs
+    /// Will need to divert from IERC1271 standard but still follow return values
+    function isValidSig(
```

Comment:
Are we gonna support BLS signatures as well?

Reply:
The only viable option for BLS that I can see is that we would need to pull all of the contracts out of the implementation, and use the directly in our repo. This is due to the fact that there are no packages available to perform the steps required for testing that can be found in 4337 (due to them testing with ethers it opens up being able to use packages for ts/js). I have tried creating my own functions to use with foundry and while a couple of them work (or at least seem to), some of the others dont and I dont understand cryptography enough to create these. Also, if we want to use BLS, it would be better to split this into a separate ticket as BLS would not just be a plugin, it would have to be a new factory, wallet and plugin.